spec/controllers/reservations_controller_spec.rb

```diff
@@ -1,247 +1,189 @@
+        UserMock.new(reservations: res)
       end
-
-      it_behaves_like 'can view reservation by patron'
-      it_behaves_like 'can view reservation by admin'
-    end
-
-    context 'when accessed by checkout person' do
       before(:each) do
-        sign_in @checkout_person
+        mock_user_sign_in(user)
+        get :index
       end
-
-      it_behaves_like 'can view reservation by patron'
-      it_behaves_like 'can view reservation by admin'
+      it { is_expected.to respond_with(:success) }
+      it { is_expected.to render_template(:index) }
+      it "only gets the current user's reservations" do
+        expect(user).to have_received(:reservations).at_least(:once)
+      end
+      it 'defaults to reserved' do
+        # twice: once from set_counts, once from filtering
+        expect(time_filtered).to have_received(:reserved).twice
+      end
+      it_behaves_like 'filterable'
     end
 
-    context 'when accessed by patron' do
+    context 'admin' do
+      let!(:user) { UserMock.new(:admin) }
       before(:each) do
-        sign_in @user
+        allow(Reservation).to receive(:starts_on_days)
+          .and_return(time_filtered)
+        mock_user_sign_in(user)
+        get :index
       end
-
-      it_behaves_like 'can view reservation by patron'
-      it_behaves_like 'cannot view reservation by admin'
+      it 'gets all reservations' do
+        expect(Reservation).to have_received(:starts_on_days)
+      end
+      it 'defaults to upcoming' do
+        # twice: once from set_counts, once from filtering
+        expect(time_filtered).to have_received(:upcoming).twice
+      end
+      it_behaves_like 'filterable'
     end
 
     it_behaves_like 'inaccessible by banned user' do
-      before { get :show, id: @reservation.id }
+      before { get :index }
     end
   end
 
   describe '#new (GET /reservations/new)' do
-    # unhappy paths: banned user, there is no reservation in the cart
+    # SMELLS:
+    #   - long method
+    #   - redirect location set for testing in code ???
+    #   - feature envy
+    # most of this should probably be in the cart?
     it_behaves_like 'inaccessible by banned user' do
       before { get :new }
     end
 
-    context 'when accessed by a non-banned user' do
+    context 'normal user' do
+      let!(:user) { FactoryGirl.build_stubbed(:user) }
       before(:each) do
-        sign_in @user
+        allow(User).to receive(:find).with(user.id).and_return(user)
+        mock_user_sign_in(user)
         request.env['HTTP_REFERER'] = 'where_i_came_from'
       end
 
       context 'with an empty cart' do
-        before(:each) do
-          get :new
-        end
+        before(:each) { get :new }
         it { expect(response).to redirect_to('where_i_came_from') }
         it { is_expected.to set_flash }
       end
 
       context 'with a non-empty cart' do
+        let!(:cart) do
+          instance_spy('Cart', items: { i: 1 }, reserver_id: user.id,
```

Comment:
I think this doesn't get caught because a cart validation would ordinarily catch it. 

In `ApplicationController`, we check if the cart is a hash, and then create a new cart object if it isn't; so here, it's a hash to get around that.

Also we aren't testing the cart contents anywhere here, so I think it's not necessarily so bad that this doesn't get caught?